src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala

```diff
@@ -254,12 +258,11 @@ object ErgoStateContext {
   }
 
   /**
-    * Initialize empty state context with fake PreHeader
+    * Initialize empty state context
     */
   def empty(genesisStateDigest: ADDigest, settings: ErgoSettings): ErgoStateContext = {
-    new ErgoStateContext(Seq.empty, None, genesisStateDigest, LaunchParameters, ErgoValidationSettings.initial, VotingData.empty)(settings.chainSettings.voting)
-      .upcoming(org.ergoplatform.mining.group.generator, 0L, settings.chainSettings.initialNBits, Array.fill(3)(0.toByte),
-        ErgoValidationSettingsUpdate.empty, 0.toByte)
+    new ErgoStateContext(Seq.empty, None, genesisStateDigest, LaunchParameters, ErgoValidationSettings.initial,
```

Comment:
Why is this, some comment for this method is required.

Reply:
This can be described as an underlying reason for the new code.
"ErgoStateContext.empty to return ErgoStateContext rather than UpcomingStateContext since it later ends up in ErgoStateContextSerializer.serialize and parse does not support it thus PreHeader was lost" can be put in the code, not in PR comments. Then you don't need to comment at all. 
Actually adding comment in the code is better, than just responding in PR comments. 
The efforts are not that larger.

Reply:
@aslesarenko 
It’s not about the efforts. My main beef with this kind of comment is that Cons outweighs the Pros.

Pros:
- you see explanation why the code was changed right beside the code change(in a PR);

Cons:
- the comment only makes sense in the context of the change (PR);
- the comment will probably confuse a future developer(without the context of the change) ; 
- the comment will inevitably be outdated since no one will update it when the code(far from this comment) is changed; 

Thus, for a one-time convenience, we'll pay too much.
I believe that the best place for a comment like that is a commit message. The second best place is PR comment.